src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,656 @@
+                                                   SyntaxEditor editor,
+                                                   CancellationToken cancellationToken)
+        {
+            var document = semanticDocument.Document;
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var methodExpression = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsMethodDeclaration(node), true)!;
+            var returnExpression = new List<SyntaxNode>
+            {
+                generator.ReturnStatement(expression.WithoutTrailingTrivia())
+            };
+
+            var typeSymbol = semanticDocument.SemanticModel.GetTypeInfo(expression, cancellationToken).Type;
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, returnType: typeSymbol);
+
+            var methodDeclaration = generator.MethodDeclaration(newMethod, returnExpression).WithAdditionalAnnotations(Formatter.Annotation);
+            editor.InsertBefore(methodExpression, methodDeclaration);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter
+        /// </summary>
+        /// 
+        private async Task IntroduceNewMethodOverloadAsync(SemanticDocument semanticDocument,
+                                                           TExpressionSyntax expression,
+                                                           IMethodSymbol methodSymbol,
+                                                           SyntaxEditor editor,
+                                                           CancellationToken cancellationToken)
+        {
+            var document = semanticDocument.Document;
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var methodExpression = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsMethodDeclaration(node), true)!;
+            var arguments = generator.CreateArguments(methodSymbol.Parameters.As<IParameterSymbol>());
+            arguments = AddExpressionArgumentToArgumentList(arguments, expression.WithoutTrailingTrivia());
+            var memberName = methodSymbol.IsGenericMethod
+                ? generator.GenericName(methodSymbol.Name, methodSymbol.TypeArguments)
+                : generator.IdentifierName(methodSymbol.Name);
+
+            var invocation = generator.InvocationExpression(memberName, arguments);
+            List<SyntaxNode> invocationReturn;
```

Comment:
why is this a list?  i only ever see a single item used.

Reply:
Because the SyntaxGenerator.MethodDeclaration requires an IEnumerable<SyntaxNode> for the statements

Reply:
my preference would be to push that to the point of where it is needed.  also, you can use SpecializedCollections.Singleton(newReturnStatement).

Reply:
this comment still applies.  also this is a list, so if we do have it, it should have a plural name.  `List<SyntaxNode> newStatements` would be fine.

Reply:
Not sure what you mean exactly by "push to the point of where it is needed", but I changed it so lmk if that is what you meant